### ISSUE TITLE
Settings.getDefaultNavigationTool() fails

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#8069]
During research for #8067 I've stumbled upon `Settings.getDefaultNavigationTool()` crashing. It tries to read a string from c:geo settings and to convert the value to an integer.

If the value returned is an empty string, converting to integer fails with an exception, leading to c:geo crashing.

For me this happened after uninstalling c:geo, reinstalling dev version and reading a backup database, thus never having set any preference. I haven't tested if it's reproducible yet, but the code should be able to cope with this conversion correctly anyway.

**Comment by bekuno** [comment]
Strange.
There is called:
~~~
getString(
                    R.string.pref_defaultNavigationTool,
                    String.valueOf(NavigationAppsEnum.COMPASS.id)));
~~~
So you schould have the String.valueOf(NavigationAppsEnum.COMPASS.id))) as default for Integer.parseInt(string).
If you get a NumberFormatException then something with getString() does not work as expected.
Maybe better to check there if sharedPrefs.getString() returns an empty string and then return the default value directly.

**Comment by moving-bits** [comment]
> Maybe better to check there if sharedPrefs.getString() returns an empty string and then return the default value directly.

IMHO `getString()` needs to be able to return an empty string for certain preferences. I created an `getIntFromString()` instead, so it's clearer what it does and can be reused. (Found a second int pref, which has been stored as string originally.) Updated my PR accordingly.

**Comment by moving-bits** [comment]
Closed with #8069
